code/data_quality_tab/data_monitoring_timeliness.R

```diff
@@ -116,62 +124,36 @@ smr04_filt <- smr04_raw%>%
-
-#seq of 2021 monthly submission deadlines. 
-#fyi: submission deadline for records to be uploaded is 6 weeks from the end
-#of the month of the clinic_date (smr00) or discharge_date(smr01, smr02, smr04)
-
-sub_deadline <- last_day_month + 42 #6 weeks is 42 days
-
-
-
-
-# Submission counts -------------------------------------------------------
+         !is.na(date_record_inserted),
+         !(hbtreat_currentdate %in% geo_excl)
+         )%>% 
+  rename("event_date" = "discharge_date")%>% 
+  mutate(smr = "SMR04")
 
+# Count Submissions ------------------------------------------
 
-submissions_smr00 <- count_submissions(smr00_filt, clinic_date, 
-                                       sub_deadline)
-submissions_smr01 <- count_submissions(smr01_filt, discharge_date,
-                                       sub_deadline)
-submissions_smr02<- count_submissions(smr02_filt, discharge_date,
-                                      sub_deadline)
-submissions_smr04 <- count_submissions(smr04_filt, discharge_date,
-                                       sub_deadline)
+# The submission dedline is 6 weeks following the end of month of discharge/transfer/death or clinic attendance 
 
-df_names <- c("submissions_smr00", "submissions_smr01",
-              "submissions_smr02", "submissions_smr04")
+#iterate through each dataframe and add a submission deadline column to the data
+for(i in 1:length(ldf)){
+  ldf[[i]] <- ldf[[i]] %>% 
+    mutate(last_day_month = lubridate::ceiling_date(event_date, "month")-1) %>% 
+    mutate(sub_deadline = last_day_month+ lubridate::days(42))
+}
 
-submissions <- append_source(df_names)%>%
-  mutate(source = case_when( source == "submissions_smr00" ~ "SMR00",
-                             source == "submissions_smr01" ~ "SMR01",
-                             source == "submissions_smr02" ~ "SMR02",
-                             TRUE ~ "SMR04")
-         )%>%
-  rename("smr" = "source")%>%
-  filter(hbres_currentdate != "S08200001", hbres_currentdate != "S08200002", 
-         hbres_currentdate != "S08200003", hbres_currentdate != "S08200004") %>% #exclude events outside Scotland
-  left_join(hb2019, by = c("hbres_currentdate"="hb"))
+# count_submissions() calculates the number of records submitted before and after the deadline
+# outputs a dataframe with counts for each SMR, HB and month
+submissions <- count_submissions(ldf) %>% 
+  left_join(hb2019, by = c("hbtreat_currentdate"="hb"))
 
 
 # Expected submissions & backlog ------------------------------------------
 
-expected_submissions_df <- read_csv(here::here("data", "expected_submissions.csv"))
+expected_submissions_df <- read_csv(
+  "/conf/Data_Quality_Dashboard/data/expected_submissions_may_sep_2021.csv")
 
 timeliness <- submissions %>% 
   left_join(expected_submissions_df)
 
-write_csv(timeliness, here::here("data", "timeliness.csv"))
+write_csv(timeliness, "/conf/Data_Quality_Dashboard/data/timeliness.csv")
```

Comment:
The file paths here will be affected by the decisions on the project structure and relative file paths.